.github/bin/prepare-check-commits-matrix.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import logging
+import sys
+import tempfile
+import unittest
+
+import csv
+import json
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Choose commits to compile using their CSV descriptions."
+    )
+    parser.add_argument(
+        "-i",
+        "--input",
+        type=argparse.FileType("r"),
+        default=sys.stdin,
+        help="A CSV file in <commit hash>,<tree hash>,<commit subject>",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="Filename to write chosen commit hashes to",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_const",
+        dest="loglevel",
+        const=logging.INFO,
+        default=logging.WARNING,
+        help="Print info level logs",
+    )
+    parser.add_argument(
+        "-t",
+        "--test",
+        action='store_true',
+        help="test this script instead of executing it",
+    )
+
+    args = parser.parse_args()
+    logging.basicConfig(level=args.loglevel)
+    if args.test:
+        sys.argv = [sys.argv[0]]
+        unittest.main()
+        return
+    build(args.input, args.output)
+
+
+def build(input_file, output_file):
+    logging.info("input_file: %s", input_file)
+    logging.info("output_file: %s", output_file)
+    reader = csv.reader(input_file)
+    entries = list(reader)
+    logging.info("entries: %s", entries)
+    commits = []
+    for i, entry in enumerate(entries):
+        try:
+            commit_hash, _, subject = entry
+            # TODO: add filtering based on GitHub Actions cache entries
+            if not has_followup(entries, i, commit_hash, subject):
+                commits.append(commit_hash)
+        except ValueError:
+            # ignore lines which don't match the expected CSV pattern
+            pass
+    if len(commits) > 0:
+        json.dump({'include': [{'commit': commit} for commit in commits]}, output_file, separators=(',', ':'), sort_keys=True)
```

Comment:
Do we want to sort the commits alphabetically or preserve their order?

Reply:
This doesn't sort lists, only keys in objects.
It makes the output deterministic, but it's not really necessary since all of the objects we generate now have only one key. 

The commits should be from oldest to newest as generated by `git log --reverse`